enterprise/views.py

```diff
@@ -479,45 +480,69 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
             raise Http404
 
         try:
+            enrollment_client = EnrollmentApiClient()
+            modes = enrollment_client.get_course_modes(course_id)
+        except HttpClientError:
+            logger.error('Failed to determine available course modes for course ID: %s', course_id)
+            raise Http404
+
+        try:
             enterprise_uuid = UUID(enterprise_uuid)
             # pylint: disable=no-member
             enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_uuid)
         except (ValueError, EnterpriseCustomer.DoesNotExist):
             logger.error('Unable to find enterprise customer for UUID: %s', enterprise_uuid)
             raise Http404
 
+        course_modes = []
+
+        audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit'])
+
+        for mode in modes:
+            if mode['min_price']:
+                price_text = '${}'.format(mode['min_price'])
+            else:
+                price_text = self.context_data['free_price_text']
+            if mode['slug'] in audit_modes:
+                description = self.context_data['audit_text']
+            else:
+                description = self.context_data['verified_text']
+            course_modes.append({
+                'mode': mode['slug'],
+                'title': mode['name'],
+                'original_price': price_text,
+                'final_price': price_text,
```

Comment:
@bdero @douglashall this functionality will be implemented as part of [ENT-357](https://openedx.atlassian.net/browse/ENT-357) on which @bradenmacdonald is working. Also I have closed `ENT-406` as duplicate of [ENT-357](https://openedx.atlassian.net/browse/ENT-357).